src/sentry/integrations/jira/notify_action.py

```diff
@@ -52,15 +55,16 @@ def after(self, event, state):
         def create_issue(event, futures):
             """Create the Jira ticket for a given event"""
 
-            # TODO check if a Jira ticket already exists for the given event's issue. if it does, skip creating it
-            resp = installation.create_issue(self.data)
-            ExternalIssue.objects.create(
-                organization_id=organization.id,
-                integration_id=integration.id,
-                key=resp["key"],
-                title=event.title,
-                description=installation.get_group_description(event.group, event),
-            )
+            linked = GroupLink.objects.filter(
+                project_id=self.project.id,
+                group_id=event.group.id,
+                linked_type=GroupLink.LinkedType.issue,
+            ).values_list("data", flat=True)
+
+            if not linked or json.loads(linked[0])["provider"] != self.provider:
```

Comment:
@ceorourke I don't think this will work if there are more than one linked issues on different integrations since you're only checking the 0th entry. I also think it might be better to query the actual `ExternalIssue`s via the `linked_id` and check that instead of `data` which is the JSON blob. And we should probably also check the `integration_id` instead of the `provider` since a user could have multile Jiras and we only want to dedupe on a pers installation basis.

Reply:
@scefali "...might be better to query the actual ExternalIssues via the linked_id" I'm a little confused here because `ExternalIssue` doesn't have a `linked_id` field, `GroupLink` does. Which are you suggesting to query?

Reply:
@ceorourke The `linked_id` is the ID of the `ExternalIssue`, yea? What I'm saying is to query the `ExternalIssues` linked to the `GroupLink` via your query. That has an `integration_id` we can check.